android/modules/media/src/java/ti/modules/titanium/media/MediaModule.java

```diff
@@ -1230,7 +1228,14 @@ public void previewImage(KrollDict options)
 		intent.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
 		TiIntentWrapper previewIntent = new TiIntentWrapper(intent);
 		String mimeType = image.getMimeType();
-		Uri imageUri = TiFileProvider.createUriFrom(f.getNativeFile());
+		Uri imageUri = null;
+
+		if (image.getNativePath() == null) {
```

Comment:
may want to consider refactoring this handling with the equivalent code in EmailDialogProxy. Basically in both instances you're trying to get a TiBlob converted to a File.

Reply:
For everyone's info, the way EmailDialogProxy is saving blobs to a temp file needs to be refactored too. It always saves blobs to the same filename. So, the last blob saved gets overwritten by the next blob saved to file.
https://jira.appcelerator.org/browse/TIMOB-25568

I recommend that we refactor our handling when addressing [TIMOB-25568](https://jira.appcelerator.org/browse/TIMOB-25568).